libgnucash/engine/Account.cpp

```diff
@@ -3552,9 +3525,9 @@ xaccAccountGetProjectedMinimumBalance (const Account *acc)
 
     priv = GET_PRIVATE(acc);
     today = gnc_time64_get_today_end();
-    for (node = g_list_last(priv->splits); node; node = node->prev)
+    for (auto it = priv->splits.rbegin(); it != priv->splits.rend(); it++)
     {
-        Split *split = static_cast<Split*>(node->data);
+        Split *split = *it;
```

Comment:
```
xaccAccountGetProjectedMinimumBalance (const Account *acc)
{
    bool seen_a_transaction{false};
    auto today{gnc_time64_get_today_end
    auto priv{GET_PRIVATE(acc);
    auto last_real_trans{std::find_if(priv->spllits.rbegin(), priv->splits.rend(), 
                                      [date](auto split)->bool {
                                          return xaccTransGetDate(xaccSplitGetParent(split) <= today);
                                      });
    return std::accumulate(priv->splits.rbegin(), last_real_trans, gnc_numeric_zero(),
                         [today, seen_a_transaction&](auto lowest, auto split)->gnc_numeric
                         {
                             if (seen_a_transaction)
                                 return gnc_numeric_compare(xaccSplitGetBalance(split), lowest) < 0) ?   
                                     xaccSplitGetBalance(split) : lowest;

                              seen_a_transaction = true;
                              return xaccSplitGetBalance(split);
                           });
}
```

Reply:
ditto refactored to be IMHO more functional